### PR TITLE
Add: add a new verify result status code

### PR DESCRIPTION
**Pull request by Jim8y**
This pr adds a new verifyresult status code.

The reason for this pr is becaus we need to precisely return the error status for the rpc service.

check here: https://github.com/neo-project/neo-modules/pull/815/files#r1446257434

### CODE REVIEW

File: src/Neo/Ledger/VerifyResult.cs [hunk 1]
```diff
@@ -28,6 +28,11 @@ public enum VerifyResult : byte
         /// </summary>
         AlreadyExists,
 
+        /// <summary>
+        /// Indicates that an <see cref="IInventory"/> with the same hash already exists in the memory pool.
+        /// </summary>
+        AlreadyInPool,
```
Comment (shargon) [inline src/Neo/Ledger/VerifyResult.cs:34]: Use it?

Comment (shargon) [reply src/Neo/Ledger/VerifyResult.cs:34]: We use `AlreadyExists` when it's in the pool

**Review by shargon — DISMISSED** [review]

File: src/Neo/Ledger/MemoryPool.cs [hunk 2]
```diff
@@ -293,7 +293,7 @@ internal VerifyResult TryAdd(Transaction tx, DataCache snapshot)
         {
             var poolItem = new PoolItem(tx);
 
-            if (_unsortedTransactions.ContainsKey(tx.Hash)) return VerifyResult.AlreadyExists;
+            if (_unsortedTransactions.ContainsKey(tx.Hash)) return VerifyResult.AlreadyInPool;
```
Comment (AnnaShaleva) [inline src/Neo/Ledger/MemoryPool.cs:296]: We also need to adjust the behaviour of `NeoSystem.ContainsTransaction`, aren't we?

Comment (AnnaShaleva) [reply src/Neo/Ledger/MemoryPool.cs:296]: https://github.com/neo-project/neo/blob/27df5ce9e45ae224258c7515d9e239a7ceb6d4a7/src/Neo/Ledger/Blockchain.cs#L340-L345
and
https://github.com/neo-project/neo/blob/27df5ce9e45ae224258c7515d9e239a7ceb6d4a7/src/Neo/NeoSystem.cs#L277-L281

Comment (shargon) [reply src/Neo/Ledger/MemoryPool.cs:296]: Good catch!

**Review by shargon — APPROVED** [review]

File: src/Neo/ContainsTransactionType.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+namespace Neo
+{
+    public enum ContainsTransactionType
+    {
+        NoExists,
```
Comment (AnnaShaleva) [inline src/Neo/ContainsTransactionType.cs:16]: s/NoExists/NotExist

File: src/Neo/Ledger/Blockchain.cs [hunk 4]
```diff
@@ -339,7 +339,11 @@ private VerifyResult OnNewExtensiblePayload(ExtensiblePayload payload)
 
         private VerifyResult OnNewTransaction(Transaction transaction)
         {
-            if (system.ContainsTransaction(transaction.Hash)) return VerifyResult.AlreadyExists;
+            switch (system.ContainsTransaction(transaction.Hash))
+            {
+                case ContainsTransactionType.ExistsInPool: return VerifyResult.AlreadyInPool;
+                case ContainsTransactionType.ExistsInLedger: return VerifyResult.AlreadyExists;
```
Comment (AnnaShaleva) [inline src/Neo/Ledger/Blockchain.cs:345]: `TestValidTransaction` should be fixed wrt this change (currently it's failing and it's expected).

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

File: src/Neo/Ledger/MemoryPool.cs (hunk 2 above)
Comment (AnnaShaleva) [reply src/Neo/Ledger/MemoryPool.cs:296]: Resolved.

**Review by shargon — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**